### PR TITLE
Feature/772/kebab case ccsh

### LINKED ISSUE #772: ccsh argument inconsistency
# Feature request

## Description

The argument description is inconsistent. Sometimes it says --projectName and sometimes --project-name.
We should agree on camelCase or kebab-case.

### PR DESCRIPTION
**Pull request by alexhunziker**
# Feature/772/kebab case ccsh

closes #772

## Description

- kebab case all arguments for the ccsh.
- remove --svn and --git options for scmlogparser (since they were ambiguous, instead --input-format should be used)
- minimal updates to docu to reflect the above changes

**BEFORE MERGING, MAKE SURE THAT THE SCRIPT TO BUILD THE DEMO MAPS IS UPDATED AS WELL**

## Definition of Done

A task/pull request will not be considered to be complete until all these items can be checked off.

- [x] **All** requirements mentioned in the issue are implemented
- [x] Does match the Code of Conduct and the Contribution file
- [x] Task has its own **GitHub issue** (something it is solving)
  - [x] Issue number is **included in the commit messages** for traceability
- [x] **Update the README.md** with any changes/additions made
- [x] **Update the CHANGELOG.md** with any changes/additions made
- [x] **Enough test coverage to ensure that uncovered changes do not break functionality**
- [x] **All tests pass**
- [x] **Descriptive pull request text**, answering:
  - What problem/issue are you fixing?
  - What does this PR implement and how?
- [x] **Assign your PR to someone** for a code review
  - This person _will be contacted **first**_ if a bug is introduced into `master`
- [x] **Manual testing** did not fail

### CODE REVIEW

**Comment by alschmut** [comment]
Are you sure to make these breaking changes for the ccsh, because for the „idealistic“ CC 2.0 we postponed those changes for more than a year until now..

**Comment by alexhunziker** [comment]
We discussed this PR in the last weekly and agreed on these changes.

Personally I belive that this is a different sort of magnitude than changing the .cc.json schema; we do not break any existing maps with this PR, the maximum impact is that custom scripts used to generate maps may need minor change (though if you use the short parameters like -o and so on, nothing changes). Since this issue is not related with updating the structure of the .cc.json files, I personally don't see why they should be dependent on each other when it comes to releases. For me this has the same impact as when we were removing most of the URL parameters; yes it is strictly speaking a change of the API, however it is not of the magnitude to require a major release nor do I therefore see a reason to delay resolving this issue.

**Comment by BridgeAR** [comment]
Just as a recommendation: I suggest to support both, camelCase and kebap-case. It should be relatively straight forward to unify the arguments and to use a single format internally.

**Comment by alschmut** [comment]
@alexhunziker okey that’s good if you deeply discussed that topic 👍 I was just concerned, you might not have addressed the impact (:

File: analysis/import/SourceCodeParser/README.md [hunk 1]
```diff
@@ -40,26 +40,26 @@ or
 or
 
 ```
-./ccsh sourcecodeparser src/test/resources -o foo.cc.json --defaultExcludes -e=something -e=/.*\.foo --f=table
+./ccsh sourcecodeparser src/test/resources -o foo.cc.json --default-excludes -e=something -e=/.*\.foo --f=table
```
Comment (NearW) [inline analysis/import/SourceCodeParser/README.md]: -f instead of --f

File: gh-pages/_docs/05-03-ccsh.md [hunk 2]
```diff
@@ -16,8 +16,6 @@ If you want to combine multiple metrics into one `.cc.json` you should get comfo
 
 Please note that most of our tool documentation still lives in the Readme and we haven't migrated all their documentation to this page. For example the SonarImporter is documented [here](https://github.com/MaibornWolff/codecharta/tree/master/analysis/import/SonarImporter).
 
-It's also worth noting that we haven't been consistent with the prefixes for our tools. Sorry. Aligning all those names would probably mean a [BREAKING Api change]({{site.baseurl}}{% link _docs/01-05-versioning.md %}) and we aren't sure that it's worth that.
```
Comment (NearW) [inline gh-pages/_docs/05-03-ccsh.md]: good catch, lol

**Pull request merged**